### PR TITLE
feat: Adding sse-kms support for Mountpoint S3 CSI driver EKS IRSA

### PR DESCRIPTION
**Pull request by philslab-ninja**
## Description
To use aws:kms encryption with the Mountpoint S3 CSI driver the IAM policy needs to include kms:Decrypt and kms:GenerateDataKey permissions for the KMS key used for the bucket(s)

## Motivation and Context
Without the change usage of Buckets with aws:kms encryption is not possible for Mountpoint S3 CSI driver 

## Breaking Changes
none

## How Has This Been Tested?
- [x] I have tested and validated the changes using my fork on existing infrastructure

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: modules/iam-role-for-service-accounts-eks/variables.tf [hunk 1]
```diff
@@ -158,6 +158,12 @@ variable "mountpoint_s3_csi_bucket_arns" {
   default     = []
 }
 
+variable "mountpoint_s3_csi_kms_cmk_ids" {
```
Comment (bryantbiggs) [inline modules/iam-role-for-service-accounts-eks/variables.tf:161]: Lets drop the `cmk` from the variable name

```suggestion
variable "mountpoint_s3_csi_kms_ids" {
```

Comment (philslab-ninja) [reply modules/iam-role-for-service-accounts-eks/variables.tf:161]: okay, I will do that.
I was copying the name of the ebs policy, where `ebs_csi_kms_cmk_ids` is used as variable name.

**Comment by philslab-ninja** [comment]
@bryantbiggs : If you have the time please review once more, I have changed the variable name as you have suggested

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 5.41.0 :tada:

**Pull request merged**